In [21]:
from Utils.setupDataset import get_dataset, combine_datasets
from Utils.feature_vector import get_feature_names

fnames = get_feature_names()
import numpy as np

from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import GridSearchCV

fnames = get_feature_names()

m_dataset = get_dataset("dataset.pickle")
w_dataset = get_dataset("wiki_dataset.pickle")

c_dataset = combine_datasets(m_dataset, w_dataset)

# Setup the hyperparameter grid
c_space = np.logspace(-5, 8, 15)
param_grid = {'C': c_space}

# Instantiate a logistic regression classifier: logreg
logreg = LogisticRegression(penalty='l1')

# Instantiate the GridSearchCV object: logreg_cv
logreg_cv = GridSearchCV(logreg, param_grid, cv=5)


X = w_dataset["data"] 
y = w_dataset["target"]
# Fit it to the data
logreg_cv.fit(X, y)

# Print the tuned parameters and score
print("Tuned Logistic Regression Parameters: {}".format(logreg_cv.best_params_)) 
print("Best score is {}".format(logreg_cv.best_score_))




#param_grid = {'C': [0.001, 0.01, 0.1, 1, 10, 100, 1000] }

#clf = GridSearchCV(LogisticRegression(penalty='l2'), param_grid)
#GridSearchCV(cv=None,
#       estimator=LogisticRegression(C=1.0, intercept_scaling=1, dual=False, fit_intercept=True,
#          penalty='l2', tol=0.0001),
#       param_grid={'C': [0.001, 0.01, 0.1, 1, 10, 100, 1000]})

Tuned Logistic Regression Parameters: {'C': 31.622776601683793}
Best score is 0.884522854852


Penalty L1:
# my dataset:
Tuned Logistic Regression Parameters: {'C': 2275.8459260747909}
Best score is 0.816247582205

Tuned Logistic Regression Parameters: {'C': 268.26957952797272}
Best score is 0.817214700193

Tuned Logistic Regression Parameters: {'C': 19306.977288832535}
Best score is 0.816247582205

# wiki dataset:
Tuned Logistic Regression Parameters: {'C': 268.26957952797272}
* Best score is 0.884121892542

# combined dataset
Tuned Logistic Regression Parameters: {'C': 31.622776601683793}
Best score is 0.857854533378

penalty L2

# my dataset:
Tuned Logistic Regression Parameters: {'C': 1389495.494373136}
Best score is 0.817214700193

* Tuned Logistic Regression Parameters: {'C': 19306.977288832535}
Best score is 0.81914893617

Tuned Logistic Regression Parameters: {'C': 100000000.0}
Best score is 0.817214700193

Tuned Logistic Regression Parameters: {'C': 19306.977288832535}
Best score is 0.818181818182

# wiki dataset:
Tuned Logistic Regression Parameters: {'C': 1389495.494373136}
Best score is 0.862670408982

* Tuned Logistic Regression Parameters: {'C': 19306.977288832535}
Best score is 0.861868484362

Tuned Logistic Regression Parameters: {'C': 1389495.494373136}
Best score is 0.861066559743

Tuned Logistic Regression Parameters: {'C': 0.051794746792312128}
Best score is 0.861868484362

# combined dataset:
Tuned Logistic Regression Parameters: {'C': 3.7275937203149381}
Best score is 0.822982397874

Tuned Logistic Regression Parameters: {'C': 31.622776601683793}
Best score is 0.824144802391

Tuned Logistic Regression Parameters: {'C': 100000000.0}
Best score is 0.823148455663

* Tuned Logistic Regression Parameters: {'C': 19306.977288832535}
Best score is 0.823978744603